proxy/proxy.go

```diff
@@ -3,82 +3,90 @@ package proxy
 import (
 	"fmt"
 	"net/url"
-	"time"
+	"os"
+	"os/signal"
+	"context"
+	"syscall"
 
-	"github.com/9seconds/httransform"
+	"github.com/9seconds/httransform/v2"
+	"github.com/9seconds/httransform/v2/auth"
+	"github.com/9seconds/httransform/v2/layers"
+	"github.com/9seconds/httransform/v2/dialers"
+	"github.com/9seconds/httransform/v2/executor"
 
 	"github.com/scrapinghub/crawlera-headless-proxy/config"
-	"github.com/scrapinghub/crawlera-headless-proxy/layers"
+	customs "github.com/scrapinghub/crawlera-headless-proxy/layers"
 	"github.com/scrapinghub/crawlera-headless-proxy/stats"
 )
 
 func NewProxy(conf *config.Config, statsContainer *stats.Stats) (*httransform.Server, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		for range signals {
+			cancel()
+		}
+	}()
```

Comment:
Maybe worth adding comment about why this code is needed? It is not very clear why it is added

Reply:
That's copy-paste from official example and I have no idea why it's needed :)
Agree, it may worth to add [this link](https://github.com/9seconds/httransform/blob/master/README.md#example) as a comment.

Reply:
This is a handler for unix signals. Once either SIGINT or SIGTERM arrive, you are getting a new message in signals channel. So, once you've got it, you call `cancel` function to terminate a context. This context termination make all child contexts to be cancelled as well.

Reply:
Moved context creation into the main, otherwise it was canceled immediately after exiting `NewProxy()`